### PR TITLE
Implement external player feature

### PR DESCRIPTION
**Pull request by CarlosOlivo**
### Implement jellyfin-archive/jellyfin-android-original#67
Feel free to refactor the source code, the companion objects, Kotlin coroutines and english as well aren't my thing yet.

Supported players
- VLC Media Player
- MX Player

Notes
- Only works with DirectStream play method since a stream url is required.
- Uses ExoPlayer device profile since there is no way to get it from an external player.

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
@Maxr1998 for reference, this is (part of) the external-player code in ATV: https://github.com/jellyfin/jellyfin-androidtv/blob/master/app/src/main/java/org/jellyfin/androidtv/playback/ExternalPlayerActivity.java

**Comment by nielsvanvelzen** [comment]
I think we should not use the "external video players" option in the webui but instead add the option to the client settings and merge it with the native video player setting:

**Prefferred player**

- Web player (default)
- ExoPlayer
- External

Right now being able to enable both native and external is weird.

File: app/src/main/java/org/jellyfin/mobile/MainActivity.kt
Comment (Maxr1998) [inline app/src/main/java/org/jellyfin/mobile/MainActivity.kt:224]: You could move that above the `onRequestPermissionResult`, I think that'd be a bit neater.

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt
Comment (Maxr1998) [inline app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt:48]: I'd suggest to pass this activity reference to the handle functions directly, to ensure that we never leak this context if in some case it isn't set to null again.

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt
Comment (Maxr1998) [inline app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt:143]: If you want, you could make this (and the other functions) extension functions of `MainActivity` to always have a context ready.

**Review by Maxr1998 — CHANGES_REQUESTED** [review]
Generally, most of the code looks quite good, have some small comments that'd need fixing though. Also, I noticed that your formatting sometimes doesn't completely follow the codestyle (we use the Kotlin official style with some customization in our editorconfig, if you open the code in Android Studio and reformat your files it should automatically apply it), e.g. there's a space missing after `if`s, and assignments are sometimes missing a space as well.

Also, I agree with @nielsvanvelzen that we should put the setting into the client settings, that could be done after this PR is merged however.

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt
Comment (CarlosOlivo) [reply app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt]: I think it is not necessary to expose a function that will only be used in ExternalPlayer, the same with some constants, considering that the video player webapp events are eliminated in #104

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt
Comment (CarlosOlivo) [reply app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt]: I changed the way to call the functions, in theory there should be no problem with context leaks now.

File: app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt
Comment (Maxr1998) [inline app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt:54]: Could you move those to the Constants file please? Below the result actions for example, with a comment like `// External player webapp events`.

**Review by Maxr1998 — CHANGES_REQUESTED** [review]
Thanks for the changes! If you fix the constants like I described and rebase your changes onto the latest master, I can merge it. Feel free to squash your commits into a single one, if you like.

**Comment by Maxr1998** [comment]
@nielsvanvelzen I'd appreciate if you could take a look over this as well, not that I miss something.

**Review by nielsvanvelzen — APPROVED** [review]
Looks fine to me, only things I'd change is a few naming choices (like externalplayer instead of extplayer) and some code style. Nothing major so you got my approval 👍

**Review by Maxr1998 — APPROVED** [review]

**Pull request merged**
